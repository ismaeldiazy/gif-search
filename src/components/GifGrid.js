import React from 'react'

export const GifGrid = ({ category }) => {
    
    const getGifs = async() => {

        const url = 'https://api.giphy.com/v1/gifs/search?q=cheeseburger&limit=10&api_key=e4Rbc8Hq9DMMjYD6rbhsFUnRdqoxGorD'
        const response = await fetch(url);
        const { data } = await response.json();

        const gifs = data.map( img => {
            return {
                id: img.id,
                title: img.title,
                url: img.images?.downsized_medium.url
            }
        });

        console.log(gifs);
    }

    getGifs();

    return (
        <div>
            <h3> { category } </h3>
        </div>
    )
}
