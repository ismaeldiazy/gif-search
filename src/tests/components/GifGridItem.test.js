import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import { GifGridItem } from '../../components/GifGridItem';



describe('Pruebas sobre el componente <GifGridItem />', () => {

    const title = 'title';
    const url = 'https://test.com';
    const wrapper = shallow(<GifGridItem title={ title } url={ url } />);

    test('should return the component', () => {
    
        expect(wrapper).toMatchSnapshot();
    })

    test('img should have url and alt like props', () => {
        const img = wrapper.find('img');

        expect(img.prop('src')).toBe(url);
        expect(img.prop('alt')).toBe(title);
    })

    test('should have animate__fadeIn class in Div', () => {
        const className = wrapper.find('div').prop('className');

        expect(className.includes('animate__fadeIn')).toBe( true );
    })
    
    
    
})


