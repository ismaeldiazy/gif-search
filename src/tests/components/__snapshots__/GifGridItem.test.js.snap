// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas sobre el componente <GifGridItem /> should return the component 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="title"
    onClick={[Function]}
    src="https://test.com"
  />
</div>
`;
